### PR TITLE
Move common SQL generation code into unified SQL generator

### PR DESCRIPTION
**Pull request by mikary**
Basic refactoring for #2493

### CODE REVIEW

**Comment by anpete** [comment]
:shipit: after interface change

File: src/EntityFramework.Relational/Migrations/HistoryRepository.cs [hunk 1]
```diff
@@ -51,14 +50,14 @@ public abstract class HistoryRepository : IHistoryRepository
...
-            Check.NotNull(sql, nameof(sql));
+            Check.NotNull(sqlGenerator, nameof(sqlGenerator));
 
             _databaseCreator = (IRelationalDatabaseCreator)databaseCreator;
             _executor = executor;
             _connection = connection;
             _modelDiffer = modelDiffer;
             _migrationsSqlGenerator = migrationsSqlGenerator;
-            Sql = sql;
+            SqlGenerator = sqlGenerator;
```
Comment (bricelam) [inline src/EntityFramework.Relational/Migrations/HistoryRepository.cs]: :frowning: Didn't we decide just *Sql* was OK in API Review?

File: src/EntityFramework.Relational/Migrations/MigrationsSqlGenerator.cs [hunk 2]
```diff
@@ -53,21 +53,26 @@ public class MigrationsSqlGenerator : IMigrationsSqlGenerator
 
         public MigrationsSqlGenerator(
             [NotNull] IRelationalCommandBuilderFactory commandBuilderFactory,
+            [NotNull] ISqlGenerator sqlGenerator,
             [NotNull] IUpdateSqlGenerator sql,
```
Comment (bricelam) [inline src/EntityFramework.Relational/Migrations/MigrationsSqlGenerator.cs]: Does this really need both or can `IUpdateSqlGenerator` be removed?

File: src/EntityFramework.Relational/Migrations/HistoryRepository.cs (hunk 1 above)
Comment (mikary) [inline src/EntityFramework.Relational/Migrations/HistoryRepository.cs]: Oops, didn't mean to change the name, this is just an artifact of some incremental changes where both generators were being used.

File: src/EntityFramework.Relational/Migrations/MigrationsSqlGenerator.cs (hunk 2 above)
Comment (mikary) [inline src/EntityFramework.Relational/Migrations/MigrationsSqlGenerator.cs]: If the referenced code on `IUpdateSqlGenerator` (or any of the other generators) is general purpose, we could move it to the `ISqlGenerator`. I expect we will want to keep adding to this as we refine SQL generation (i.e. parameter prefixes for #2916, operators, etc.)

**Pull request merged**